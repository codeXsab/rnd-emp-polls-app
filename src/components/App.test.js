import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render } from "@testing-library/react";
import App from "./App";
import { legacy_createStore as createStore } from "redux";
import { Provider } from "react-redux";
import rootReducer from "../reducers/index";
import middleware from "../middleware/index";
import { setAuthedUser} from "../actions/authedUser";
import "@testing-library/jest-dom";
const store = createStore(rootReducer, middleware);

describe("App", () => {
    it("should render the component", () => {
        const component = render(
            <Provider store={store}>
                <Router>
                    <App/>
                </Router>
            </Provider>
        );
        expect(component).toBeDefined();
        expect(component).toMatchSnapshot();
    });

    it("should show Login page when not logged in", () => {
        const component = render(
            <Provider store={store}>
                <Router>
                    <App/>
                </Router>
            </Provider>
        );
        const heading = component.getByTestId("login-form");
        expect(heading).toBeInTheDocument();
    });

    it("should show Dashboard page when logged in", () => {
        store.dispatch(setAuthedUser({ id: "sarahedo" }));

        const component = render(
            <Provider store={store}>
                <Router>
                    <App/>
                </Router>
            </Provider>
        );

        const heading = component.getByTestId("navbar");
        expect(heading).toBeInTheDocument();
    });
});