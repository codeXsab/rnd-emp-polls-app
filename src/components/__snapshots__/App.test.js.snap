// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid"
      >
        <div />
        <div
          class="justify-contents-center align-items-center m-auto pt-5 row"
        >
          <div
            class="m-auto col-md-6 col-12"
          >
            <div
              class="bg-light text-center mt-5 border-info card"
            >
              <div
                class="p-5 bg-info card-header"
              >
                <h1
                  class="text-center fw-bold"
                >
                  Employee Polls Application
                </h1>
                <h3
                  class="text-center p-2"
                >
                  Please sign in to continue.
                </h3>
              </div>
              <img
                alt="company logo"
                class="w-80 p-5 img-fluid"
                src="18940.jpg"
              />
              <div
                class="card-body"
              >
                <form
                  class="Form"
                  data-testid="login-form"
                >
                  <div>
                    <select
                      aria-label="Select a User"
                      class="p-2 form-control form-control-lg"
                      data-testid="login-select"
                    >
                      <option
                        class="user-dropdown"
                        disabled=""
                        value="none"
                      >
                        Select a User...
                      </option>
                    </select>
                    <button
                      class="w-100 mt-3 btn-lg btn-info btn btn-primary"
                      data-testid="submit"
                      disabled=""
                      type="button"
                    >
                      <span
                        class="fw-bold"
                      >
                        Sign In
                      </span>
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid"
    >
      <div />
      <div
        class="justify-contents-center align-items-center m-auto pt-5 row"
      >
        <div
          class="m-auto col-md-6 col-12"
        >
          <div
            class="bg-light text-center mt-5 border-info card"
          >
            <div
              class="p-5 bg-info card-header"
            >
              <h1
                class="text-center fw-bold"
              >
                Employee Polls Application
              </h1>
              <h3
                class="text-center p-2"
              >
                Please sign in to continue.
              </h3>
            </div>
            <img
              alt="company logo"
              class="w-80 p-5 img-fluid"
              src="18940.jpg"
            />
            <div
              class="card-body"
            >
              <form
                class="Form"
                data-testid="login-form"
              >
                <div>
                  <select
                    aria-label="Select a User"
                    class="p-2 form-control form-control-lg"
                    data-testid="login-select"
                  >
                    <option
                      class="user-dropdown"
                      disabled=""
                      value="none"
                    >
                      Select a User...
                    </option>
                  </select>
                  <button
                    class="w-100 mt-3 btn-lg btn-info btn btn-primary"
                    data-testid="submit"
                    disabled=""
                    type="button"
                  >
                    <span
                      class="fw-bold"
                    >
                      Sign In
                    </span>
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
